### DISCUSSION TITLE
How do i get all posts instead of last 10?

### DISCUSSION THREAD

**Discussion by ranababu1** [wpengine/faustjs#381] [Q&A]
First of all, thank you for this amazing framework.
My query is how do I get all the posts from my WordPress blog instead of the last 10 and paginate them 10 posts per page exactly how WordPress natively does.

**Comment by josephfusco** [comment]
Hey @ranababu1 thank you for your kind words!

Is this still an issue for you? Faust has evolved a lot since you initial had this question.

**Reply by SimoCharhbili** [reply]
I have just started using Faust, and I am looking for a solution for this issue, I would appreciate it if you share a solution to this matter.

**Reply by treithamer** [reply]
limit the # of posts
You may take a look at index.tsx, line 14, 
https://github.com/wpengine/faustjs/blob/canary/examples/next/getting-started/src/pages/index.tsx

Here's usePosts Hook
https://faustjs.org/docs/next/reference/hooks/usePosts

Here's and example on pagination
https://github.com/wpengine/faustjs/blob/canary/examples/next/getting-started/src/pages/posts/index.tsx#L18-L23
